### PR TITLE
Add Validation for IAP domain and allowlist

### PR DESCRIPTION
**Pull request by umeshkumhar**
- Add checks that will fail fast, if domain and member_allowlist are empty..

### CODE REVIEW

File: modules/iap/iap.tf [hunk 1]
```diff
@@ -21,6 +21,23 @@ data "google_project" "project" {
...
 }
 
+## Check if domain and members_allowlist is provided
+data "null_data_source" "domain_validation" {
+  inputs = {}
+
+  lifecycle {
+    precondition {
+      condition     = var.domain != ""
+      error_message = "IAP configured requires domain name, Please provide a valid domain name for ${var.app_name} application."
```
Comment (imreddy13) [inline modules/iap/iap.tf:31]: "IAP configuration"

File: modules/iap/iap.tf [hunk 2]
```diff
@@ -21,6 +21,23 @@ data "google_project" "project" {
...
+    }
+
+    precondition {
+      condition     = length(var.members_allowlist) != 0
+      error_message = "IAP configured requires valid members_allowlist, Please provide a valid input for ${var.app_name} application."
```
Comment (imreddy13) [inline modules/iap/iap.tf:36]: "IAP configuration requires allowlisting users. Please provide a valid allowlist for ${var.app_name} application."

**Review by imreddy13 — APPROVED** [review]

File: modules/iap/iap.tf [hunk 3]
```diff
@@ -21,6 +21,23 @@ data "google_project" "project" {
...
+  inputs = {}
+
+  lifecycle {
+    precondition {
+      condition     = var.domain != ""
+      error_message = "IAP configuration requires domain name, Please provide a valid domain name for ${var.app_name} application."
+    }
+
+    precondition {
+      condition     = length(var.members_allowlist) != 0
```
Comment (blackzlq) [inline modules/iap/iap.tf:35]: member allowlist can be empty, they have the right to add them later

Comment (imreddy13) [reply modules/iap/iap.tf:35]: I think this just makes the experience more confusing, most users would want to allowlist at least themselves - they'll leave this blank and wonder why they can't login later. Prefer to fail early and force them to add something to the allowlist.

**Comment by andrewsykim** [comment]
LGTM

**Pull request merged**